active/002-py3-only.md

```diff
@@ -0,0 +1,72 @@
+---
+Feature name: py3-only
+Start date: 2015-09-23
+Pull request: <leave empty>
+Authors: ["@jairideout", "@gregcaporaso"]
+Contributors:
+-
+---
+
+# Summary
+
+Remove Python 2 (Legacy Python) support from scikit-bio in favor of Python 3
+support only.
+
+# Motivation
+
+We propose removing Python 2 (Legacy Python) support from scikit-bio for the
+following reasons:
+
+1. The scientific Python community is in favor of dropping support for Python 2
+as soon as possible (e.g., see [recent discussion at DS4DS](https://bids.hackpad.com/Planning-an-early-death-for-Python-2-Legacy-Python.-6aB7agi5XCr)).
+2. Dropping Python 2 support will reduce development and maintenance burden on
+scikit-bio developers.
+3. Supporting Python 3 only will allow us to start using new features of the
+language, ultimately making scikit-bio a better package.
+4. Given that scikit-bio is still in beta and has relatively few users (e.g.,
+compared to a package like numpy), making this switch now would be better than
+waiting and could set a precedent for other scientific Python packages.
+
+# Detailed design
+
+Changes to support Python 3-only include:
+
+1. Remove Python 2 tests from Travis-CI build matrix.
+2. Remove `future` and `six` dependencies from scikit-bio.
+3. Update documentation to indicate Python 3-only, including (but not limited
+to) `README.md` and `CONTRIBUTING.md`.
+4. Remove `doc/source/development/py3.rst`.
+5. Update PyPI classifiers in `setup.py` to indicate Python 3-only.
+6. Search codebase for references to Python 2 and remove/update accordingly.
+This includes comments and actual code modifications.
+7. Removes hacks in `skbio.io` necessary for Python 2 compatibility.
+8. Test against latest release and nightly build of Python 3 on Travis-CI.
+Allow failure against nightly Python 3 build.
+
+Note: scikit-bio's doctests are already Python 3-only, so no updates necessary
+there! :shipit:
+
+# Drawbacks
+
+Packages/tools supporting Python 2 (including those supporting Python 2
+**and** 3) will not be able to depend on scikit-bio.
```

Comment:
The major place I see this impacting the current development is Qiime. This may not be relevant, because it appears that QIIME 1.9.1 relies on Scikit-bio 0.2.3 and Python 2.
It may create challenges for users who wish to have a QIIME installation and a later scikit-bio installation but do not have virtualenvs or conda. 

So, I guess a piece of the question about the timeline for transition is deals with the timeline for QIIME 2.0, and the plans for that software stack.